aten/src/ATen/native/mps/operations/ReduceOps.mm

```diff
@@ -245,9 +246,21 @@ static void reduction_out_mps(const Tensor& input_t,
 
         castOutputTensor = [mpsGraph reductionSumWithTensor:nonZeros axes:wrappedAxes name:nil];
       } else if (reduction_type == MPSReductionType::AMAX) {
-        castOutputTensor = [mpsGraph reductionMaximumWithTensor:castInputTensor axes:wrappedAxes name:nil];
+        if (macOS15_0_plus) {
+          castOutputTensor = [mpsGraph reductionMaximumPropagateNaNWithTensor:castInputTensor
+                                                                         axes:wrappedAxes
+                                                                         name:nil];
+        } else {
+          castOutputTensor = [mpsGraph reductionMaximumWithTensor:castInputTensor axes:wrappedAxes name:nil];
+        }
       } else if (reduction_type == MPSReductionType::AMIN) {
-        castOutputTensor = [mpsGraph reductionMinimumWithTensor:castInputTensor axes:wrappedAxes name:nil];
+        if (macOS15_0_plus) {
```

Comment:
Hmm, why limit to MacOS 15 here, if according to the documentation this method exists since MacOS-11:
https://developer.apple.com/documentation/metalperformanceshadersgraph/mpsgraph/reductionminimumpropagatenanwithtensor:axes:name:

Reply:
I'm pretty sure the documentation is lying here. I'll try to ping the relevant parties to update that page but at least based on local testing and me having filed the request for that particular API not too long ago I'd be surprised if that was the case.

Reply:
Ok actually the doc is not that wrong. I'm being told it went into MacOS12. I'll need to find a Macos14 machine to verify why I saw an issue there in that case and adjust the availability accordingly.

Reply:
Ok the missing selector complaint I was seeing on MacOS14 was due to one NaN being spelled as Nan. My brain just refused to register that. I've now removed the limitations since the selector should be available from MacOS12 onwards so it should be supported across the OS versions we've said work with PyTorch. Good call.